### PR TITLE
[bitstreams] Splice bazel-built ROMs by default

### PR DESCRIPTION
**Pull request by a-will**
Change the default to use bazel-built ROMs for bitstreams instead of
whatever was in the bitstream when it was uploaded to the cache. This
makes the default bazel flow use the ROMs that are part of the current
source tree and avoids relying on a particular variant being available
in the cache.

As a side effect, software development on the FPGA now firmly requires
the FPGA lab tools by default. Update the documentation to reflect the
new behavior and tool requirements.

Also fix up the fusesoc rule to use the bazel-provided python interpreter.

### CODE REVIEW

**Comment by a-will** [comment]
> bazel build --distdir=bazel-airgapped/bazel-distdir --repository_cache=bazel-airgapped/bazel-cache <label>
> + trap remove_airgapped_netns EXIT
> + sudo ip netns add airgapped
> mount --make-shared /var/run/netns failed: Permission denied
> + remove_airgapped_netns
> + sudo ip netns delete airgapped
> Cannot remove namespace file "/var/run/netns/airgapped": No such file or directory

Hm... with the change to the ci-public pool (to gain access to a Vivado installation), it looks like we might be in a container and can't create a network namespace for testing air-gapped builds.

File: doc/getting_started/setup_fpga.md [hunk 1]
```diff
@@ -41,8 +41,8 @@ tar -xvf bitstream-latest.tar.gz
 
 By default, the bitstream is built with a version of the boot ROM used for testing (called the _test ROM_; pulled from `sw/device/lib/testing/test_rom`).
 There is also a version of the boot ROM used in production (called the _ROM_; pulled from `sw/device/silicon_creator/rom`).
-This can be spliced into the bitstream to overwrite the testing boot ROM as described in the [FPGA Reference Manual]({{< relref "ref_manual_fpga.md#boot-rom-development" >}}).
-However, if you do not want to do the splicing yourself, both versions of the bitstream are available in the download as `*.bit.orig` and `*.bit.splice` (containing the test ROM and the ROM respectively).
+When the bitstream cache is used in bazel flows, the ROMs from the cache are not used.
+Instead, the bazel-built ROMs are spliced into the image and create new bitstreams, using the mechanism described in the [FPGA Reference Manual]({{< relref "ref_manual_fpga.md#boot-rom-development" >}}).
```
Comment (timothytrippel) [inline doc/getting_started/setup_fpga.md:45]:
```suggestion
Instead, the bazel-built ROMs are spliced into the image to create new bitstreams, using the mechanism described in the [FPGA Reference Manual]({{< relref "ref_manual_fpga.md#boot-rom-development" >}}).
```

File: doc/getting_started/install_vivado/index.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ Generating a bitstream for Xilinx devices requires a
 Please note that the "WebPACK" edition __does not__ support the Xilinx Kintex 7
 XC7K410T used on the CW310 board.
 
+For software development, Vivado is still necessary for most workflows.
+However, the Lab Edition is sufficient, and it has a significantly smaller installation footprint.
```
Comment (timothytrippel) [inline doc/getting_started/install_vivado/index.md:11]:
```suggestion
However, the (free) Lab Edition is sufficient, and it has a significantly smaller installation footprint.
```
?

**Review by timothytrippel — APPROVED** [review]

**Comment by timothytrippel** [comment]
I believe @a-will mentioned this offline, but just to confirm / for transparency, once this goes in, there will we be a follow up PR that removes one of the two cached bitstream ROM variants? (So there will only ever be one bitstream in the cache at each commit hash, which will act as a shell to splice in bazel built ROMs / OTP images during every build.)

**Comment by a-will** [comment]
> I believe @a-will mentioned this offline, but just to confirm / for transparency, once this goes in, there will we be a follow up PR that removes one of the two cached bitstream ROM variants? (So there will only ever be one bitstream in the cache at each commit hash, which will act as a shell to splice in bazel built ROMs / OTP images during every build.)

Yes, there will be a follow-up PR that changes the cache to only have one bitstream representing a given design. I plan to first introduce the manifest format and handling for the bitstream cache, then change how we emit cache entries.

**Comment by a-will** [comment]
I guess the sync pulled in more than our VM could handle for the air-gapped prep...

```
2023-01-11T01:50:19.4918388Z INFO: Repository remotejdk11_linux_s390x instantiated at:
2023-01-11T01:50:19.4919952Z   /DEFAULT.WORKSPACE.SUFFIX:90:6: in <toplevel>
2023-01-11T01:50:19.4921266Z   /home/vsts/.cache/bazel/_bazel_vsts/5c27df7f718ab35828b0e0cfe0993663/external/bazel_tools/tools/build_defs/repo/utils.bzl:233:18: in maybe
2023-01-11T01:50:19.4922661Z   /home/vsts/.cache/bazel/_bazel_vsts/5c27df7f718ab35828b0e0cfe0993663/external/bazel_tools/tools/jdk/remote_java_repository.bzl:48:17: in remote_java_repository
2023-01-11T01:50:19.4923532Z Repository rule http_archive defined at:
2023-01-11T01:50:19.4924203Z   /home/vsts/.cache/bazel/_bazel_vsts/5c27df7f718ab35828b0e0cfe0993663/external/bazel_tools/tools/build_defs/repo/http.bzl:353:31: in <toplevel>
2023-01-11T01:50:20.2878467Z ##[error]No space left on device
```

**Review by milesdai — APPROVED** [review]

**Pull request merged**